avro-fastserde/src/jmh/java/com/linkedin/avro/fastserde/FastAvroSerdesBenchmark.java

```diff
@@ -81,30 +98,46 @@ public void prepare() throws Exception {
     generatedRecord = (GenericData.Record) generator.generate(properties);
     serializedBytes = serializeGeneratedRecord(generatedRecord);
 
-    serializer = new AvroGenericSerializer(benchmarkSchame);
-    fastSerializer = new AvroGenericSerializer(new FastGenericDatumWriter<>(benchmarkSchame));
-    deserializer = new AvroGenericDeserializer<>(new GenericDatumReader<>(benchmarkSchame));
-    fastDeserializer = new AvroGenericDeserializer<>(new FastGenericDatumReader<>(benchmarkSchame));
+    serializer = new AvroGenericSerializer(benchmarkSchema);
+    fastSerializer = new AvroGenericSerializer(new FastGenericDatumWriter<>(benchmarkSchema));
   }
 
   @Benchmark
-  public void testAvroSerialization() throws Exception {
-    // use vanilla avro 1.4 encoder, do not use buffer binary encoder
-    serializer.serialize(generatedRecord);
+  @OperationsPerInvocation(NUMBER_OF_OPERATIONS)
+  public void testAvroSerialization(Blackhole bh) throws Exception {
+    for (int i = 0; i < NUMBER_OF_OPERATIONS; i++) {
+      // use vanilla avro 1.4 encoder, do not use buffer binary encoder
+      bh.consume(serializer.serialize(generatedRecord));
+    }
   }
 
   @Benchmark
-  public void testFastAvroSerialization() throws Exception {
-    fastSerializer.serialize(generatedRecord);
+  @OperationsPerInvocation(NUMBER_OF_OPERATIONS)
+  public void testFastAvroSerialization(Blackhole bh) throws Exception {
+    for (int i = 0; i < NUMBER_OF_OPERATIONS; i++) {
+      bh.consume(fastSerializer.serialize(generatedRecord));
+    }
   }
 
   @Benchmark
-  public void testAvroDeserialization() throws Exception {
-    deserializer.deserialize(serializedBytes);
+  @OperationsPerInvocation(NUMBER_OF_OPERATIONS)
+  public void testAvroDeserialization(Blackhole bh) throws Exception {
+    testDeserialization(new GenericDatumReader<>(benchmarkSchema), bh);
```

Comment:
should the readers be created in prepare?

Reply:
Sure, will do.